### PR TITLE
feat:add footer & nav-bar & org-card & modify theme-switcher

### PR DESCRIPTION
**Pull request by mozwell**
# What's New
- Add footer & nav-bar & org-card
- Modify theme-switcher

### CODE REVIEW

File: components/theme-switcher.tsx [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
-      }
-    >
-      {['light', 'dark', 'auto'].map((mode) => (
-        <Select.Option key={mode} value={mode}>
-          {mode}
-        </Select.Option>
-      ))}
-    </Select>
+    <Dropdown>
+      <Button variant="outline" shape="circle">
```
Comment (renzholy) [inline components/theme-switcher.tsx:21]: should close dropdown if click button when menu is open

File: components/theme-switcher.tsx [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
 import { persistentThemeAtom } from '../src/atoms'
 
+const DEFAULT_THEMES = ['light', 'dark', 'auto']
+
+function capitalize(str: string) {
+  return str.charAt(0).toUpperCase() + str.slice(1)
+}
+
 export default function ThemeSwitcher() {
   const [persistentTheme, setPersistentTheme] = useAtom(persistentThemeAtom)
```
Comment (renzholy) [inline components/theme-switcher.tsx:14]: useSetAtom

File: components/theme-switcher.tsx [hunk 3]
```diff
@@ -1,22 +1,36 @@
...
+  const handleChangeThemeGen = (theme: 'light' | 'dark' | 'auto') => () => {
```
Comment (renzholy) [inline components/theme-switcher.tsx:15]: useCallback

Comment (renzholy) [inline components/theme-switcher.tsx:15]: this kind of *Gen function seems like meaningless.

File: components/nav-bar.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+const ConnectButtonCustom = dynamic(
+  () =>
+    import('@rainbow-me/rainbowkit').then(
+      ({ ConnectButton }) => ConnectButton.Custom,
+    ),
+  { ssr: false },
+)
+
+function NavBar(props: NavBarProps) {
```
[the remark this answers is not in the document]
Comment (mozwell) [reply components/nav-bar.tsx:15]: Done.

File: components/theme-switcher.tsx [hunk 5]
```diff
@@ -1,22 +1,36 @@
 import { useAtom } from 'jotai'
-import { Select } from 'react-daisyui'
+import { Dropdown, Button } from 'react-daisyui'
+import { Theme } from '@icon-park/react'
+
 import { persistentThemeAtom } from '../src/atoms'
 
+const DEFAULT_THEMES = ['light', 'dark', 'auto']
+
+function capitalize(str: string) {
```
[the remark this answers is not in the document]
Comment (mozwell) [reply components/theme-switcher.tsx]: Fixed.

File: components/theme-switcher.tsx [hunk 6]
```diff
@@ -1,22 +1,36 @@
...
-    </Select>
+    <Dropdown>
+      <Button variant="outline" shape="circle">
+        <Theme />
+      </Button>
+      <Dropdown.Menu className="w-52">
+        {DEFAULT_THEMES.map((theme) => (
+          <Dropdown.Item
+            key={theme}
+            onClick={handleChangeThemeGen(theme as 'light' | 'dark' | 'auto')}
```
[the remark this answers is not in the document]
Comment (mozwell) [reply components/theme-switcher.tsx]: Won't fix since it is underlying logic behind CSS framework

File: components/theme-switcher.tsx (hunk 2 above)
Comment (mozwell) [reply components/theme-switcher.tsx]: Fixed.

File: components/theme-switcher.tsx (hunk 3 above)
Comment (mozwell) [reply components/theme-switcher.tsx]: Fixed.

File: components/footer.tsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
+export default function Footer() {
+  return (
+    <footer className="footer footer-center p-10 bg-base-200 text-base-content rounded">
+      <div className="grid grid-flow-col gap-4">
+        <a className="link link-hover">About</a>
+        <a className="link link-hover">Docs</a>
+        <a className="link link-hover">Twitter</a>
+      </div>
+      <div>
+        <p>Copyright © 2023 - All right reserved by Voty.xyz</p>
```
Comment (renzholy) [inline components/footer.tsx:10]: new Date().getFullYear()

File: components/theme-switcher.tsx [hunk 8]
```diff
@@ -1,22 +1,46 @@
...
 import { persistentThemeAtom } from '../src/atoms'
 
+const DEFAULT_THEMES = ['light', 'dark', 'auto']
+
 export default function ThemeSwitcher() {
-  const [persistentTheme, setPersistentTheme] = useAtom(persistentThemeAtom)
+  const setPersistentTheme = useSetAtom(persistentThemeAtom)
+
+  // To make sure clicking button will make the menu closed if the menu is open.
+  const checkAndCloseDropDown = (e: any) => {
```
Comment (renzholy) [inline components/theme-switcher.tsx:14]:
```
const handleCheckAndCloseDropDown = useCallback(e: React.MouseEvent<HTMLButtonElement>) => {
```

File: package.json [hunk 9]
```diff
@@ -22,6 +22,7 @@
     "dotbit": "^0.4.14",
     "ethers": "^5.7.2",
     "jotai": "^1.12.1",
+    "lodash": "^4.17.21",
```
Comment (renzholy) [inline package.json:25]: lodash-es already installed

File: components/theme-switcher.tsx [hunk 10]
```diff
@@ -1,22 +1,47 @@
-import { useAtom } from 'jotai'
-import { Select } from 'react-daisyui'
+import { useCallback } from 'react'
+import { useSetAtom } from 'jotai'
+import { Dropdown, Button } from 'react-daisyui'
+import { Theme } from '@icon-park/react'
+import { capitalize } from 'lodash'
```
Comment (renzholy) [inline components/theme-switcher.tsx:5]: lodash-es

File: components/theme-switcher.tsx [hunk 11]
```diff
@@ -1,22 +1,47 @@
...
 import { persistentThemeAtom } from '../src/atoms'
 
+const DEFAULT_THEMES = ['light', 'dark', 'auto']
+
 export default function ThemeSwitcher() {
-  const [persistentTheme, setPersistentTheme] = useAtom(persistentThemeAtom)
+  const setPersistentTheme = useSetAtom(persistentThemeAtom)
+
+  // To make sure clicking button will make the menu closed if the menu is open.
+  const handleCheckAndCloseDropDown = useCallback((e: any) => {
```
Comment (renzholy) [inline components/theme-switcher.tsx:15]: e: React.MouseEvent<HTMLButtonElement>

**Review by renzholy — APPROVED** [review]

**Pull request merged**